src/App.tsx

```diff
@@ -17,18 +37,32 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todos={todosFromServer}
+                handleFilterChange={setFilteredTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              {!isLoading && (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodoId={selectedTodoId}
+                  handleSelectTodoId={setSelectedTodoId}
+                />
```

Comment:
Check that the `TodoList` component correctly handles the `handleSelectTodoId` function to update the `selectedTodoId` state. This is important for selecting and displaying the correct todo in the `TodoModal`.